types/seat/wlr_seat_pointer.c

```diff
@@ -339,26 +340,33 @@ void wlr_seat_pointer_notify_motion(struct wlr_seat *wlr_seat, uint32_t time,
 uint32_t wlr_seat_pointer_notify_button(struct wlr_seat *wlr_seat,
 		uint32_t time, uint32_t button, enum wlr_button_state state) {
 	clock_gettime(CLOCK_MONOTONIC, &wlr_seat->last_event);
+
+	struct wlr_seat_pointer_state* pointer_state = &wlr_seat->pointer_state;
+
 	if (state == WLR_BUTTON_PRESSED) {
-		if (wlr_seat->pointer_state.button_count == 0) {
-			wlr_seat->pointer_state.grab_button = button;
-			wlr_seat->pointer_state.grab_time = time;
+		if (pointer_state->button_count == 0) {
+			pointer_state->grab_button = button;
+			pointer_state->grab_time = time;
 		}
-		wlr_seat->pointer_state.button_count++;
+		set_add(pointer_state->buttons, &pointer_state->button_count,
+			WLR_POINTER_BUTTONS_CAP, button);
+	/* } else if (state == WLR_BUTTON_RELEASED) { */
```

Comment:
Remnant from a debugging session?

Reply:
Yeah, I also couldn't decide which version was more correct, not that it should matter